react_project/src/components/AllProducts.jsx

```diff
@@ -1,27 +1,102 @@
+import Button from './Button';
+import styles from '../styles/AllProducts.module.css';
+
+const getPageSize = () => {
+  const width = window.innerWidth;
+  if (width < 768) return 4;
+  else if (width < 1280) return 6;
+  else return 10;
+}
+
+function AllProducts() {
+  // 데이터 받아오기
+  const [orderBy, setOrderBy] = useState("recent");
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(getPageSize());
+  const [productList, setProductList] = useState([]);
+  
+  const fetchSortedData = async ({ orderBy, page, pageSize }) => {
+    const products = await getProducts({ orderBy, page, pageSize });
+    setProductList(products.list);
+    setTotalPageNum(Math.ceil(products.totalCount / pageSize));
+  };
+
+  const handleSortSelect = (sortOption) => {
+    setOrderBy(sortOption);
+    setIsDropdownVisible(false);
+  };
+
+  useEffect(() => {
+    const handleResize = () => {
+      setPageSize(getPageSize());
+    };
+
+    window.addEventListener('resize', handleResize);
+    fetchSortedData({ orderBy, page, pageSize });
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [orderBy, page, pageSize]);
+
+  // 정렬 기준
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const toggleDropdown = () => {
+    setIsDropdownVisible(!isDropdownVisible);
+  };
+
+  // pagination
+  const [totalPageNum, setTotalPageNum] = useState();
+
+  const onPageChange = (pageNumber) => {
+    setPage(pageNumber);
+  };
 
-export default function AllProducts({ products, handleNewstClick, handleFavoriteClick }) {
   return (
-    <div className={styles.wrap}>
-      <div className={styles.products_nav}>
-        <h3>전체 상품</h3>
-        <input type="text" placeholder='검색할 상품을 등록해주세요'/>
-        <Button>상품 등록하기</Button>
-        <select name="" id="">
-          <option value="">최신순</option>
-          <option value="">좋아요순</option>
-        </select>
+    <div className={styles.allProducts_wrap}>
+      <div className={styles.allProducts_header}>
+        <h1>판매 중인 상품</h1>
+  
+        <div className={styles.allProducts_function}>
+          <div className={styles.allProducts_search}>
+            <input type="text" placeholder='검색할 상품을 입력해 주세요'>
+            </input>
+          </div>
+          <Button>상품 등록하기</Button>
+          <div>
+            <button onClick={toggleDropdown}>
+              <img src={sortIcon} alt="정렬 버튼" />
+            </button>
+            {isDropdownVisible && (
+              <DropdownList onSortSelect={handleSortSelect} />
+            )}
+          </div>
+        </div>
       </div>
-      <div className={styles.productList_container}>
-        {products.map((product) => {
-          return (
-            <div key={product.id}><Products product={product} /></div>
-          )
-        })}
+
+      <div className={styles.allProducts_list}>
+        {productList?.map((item) => (
+          <ProductCard item={item} key={`merket_item_${item.id}`} />
```

Comment:
key는 단순히 item.id 로 설정을 해주셔도 좋을 듯 괜찮을듯합니다~!
```suggestion
          <ProductCard item={item} key={item.id} />
```